dll/win32/shell32/wine/shellpath.c

```diff
@@ -106,6 +106,127 @@ DoGetProductType(PNT_PRODUCT_TYPE ProductType)
 	########## Combining and Constructing paths ##########
 */
 
+/* @implemented */
+static BOOL WINAPI
+PathSearchOnExtensionsW(LPWSTR path, LPCWSTR *dirs, BOOL flag, DWORD dwWhich)
+{
+    if (*PathFindExtensionW(path) != 0)
+        return FALSE;
+
+    if (flag)
+        return PathFindOnPathExW(path, dirs, dwWhich);
+    else
+        return PathFileExistsDefExtW(path, dwWhich);
+}
+
+/* #define REQUIREABSOLUTE */
+
+#ifdef REQUIREABSOLUTE
+/* @implemented */
+static BOOL WINAPI PathIsAbsoluteW(LPCWSTR path)
+{
+    return PathIsUNCW(path) || (PathGetDriveNumberW(path) != -1 && path[2] == L'\\');
+}
+
+/* @implemented */
+static BOOL WINAPI PathMakeAbsoluteW(LPWSTR path)
+{
+    WCHAR path1[MAX_PATH];
+    DWORD cch;
+
+    if (path == NULL)
+        return FALSE;
+    cch = GetCurrentDirectoryW(MAX_PATH, path1);
+    if (!cch || cch > MAX_PATH)
+        return FALSE;
+    return (PathCombineW(path, path1, path) != NULL);
+}
+#endif /* def REQUIREABSOLUTE */
+
+static DWORD GetShortPathNameAbsentW(LPCWSTR pszLong, LPWSTR pszShort, DWORD cchShort)
```

Comment:
Is that a new export?

Reply:
Nope. It's a function I made up.

Reply:
`GetShortPathNameW` fails if the path didn't exist.
`GetShortPathNameAbsentW` function is designed so that the system can get a valid future pathname that doesn't exist.

Reply:
GetShortPathName will also always fail on filesystems that do not support short paths.

Reply:
Added a comment.